test_ros2cli/test/ros2topic_test_configuration.py

```diff
@@ -0,0 +1,504 @@
+                "int16_values: '<sequence type: int16[3], length: 0>'",
+                "uint16_values: '<sequence type: uint16[3], length: 0>'",
+                "int32_values: '<sequence type: int32[3], length: 0>'",
+                "uint32_values: '<sequence type: uint32[3], length: 0>'",
+                "int64_values: '<sequence type: int64[3], length: 0>'",
+                "uint64_values: '<sequence type: uint64[3], length: 0>'",
+                "string_values: '<sequence type: string[3], length: 0>'",
+                "basic_types_values: '<sequence type: test_msgs/msg/BasicTypes[3], length: 0>'",
+                "constants_values: '<sequence type: test_msgs/msg/Constants[3], length: 0>'",
+                "defaults_values: '<sequence type: test_msgs/msg/Defaults[3], length: 0>'",
+                "bool_values_default: '<sequence type: boolean[3], length: 3>'",
+                "byte_values_default: '<sequence type: octet[3], length: 3>'",
+                "char_values_default: '<sequence type: uint8[3], length: 3>'",
+                "float32_values_default: '<sequence type: float[3], length: 3>'",
+                "float64_values_default: '<sequence type: double[3], length: 3>'",
+                "int8_values_default: '<sequence type: int8[3], length: 3>'",
+                "uint8_values_default: '<sequence type: uint8[3], length: 3>'",
+                "int16_values_default: '<sequence type: int16[3], length: 3>'",
+                "uint16_values_default: '<sequence type: uint16[3], length: 3>'",
+                "int32_values_default: '<sequence type: int32[3], length: 3>'",
+                "uint32_values_default: '<sequence type: uint32[3], length: 3>'",
+                "int64_values_default: '<sequence type: int64[3], length: 3>'",
+                "uint64_values_default: '<sequence type: uint64[3], length: 3>'",
+                "string_values_default: '<sequence type: string[3], length: 3>'",
+                'alignment_check: 0',
+                '---'
+            ]),
+            output_filter=basic_output_filter(
+                filtered_rmw_implementation=rmw_implementation
+            ),
+            self_terminates=False,
+            exit_codes=[2],
+            timeout=5.0
+        ),
+        CLITestConfiguration(
+            command='topic',
+            arguments=['echo', '--truncate-length', '5', '/my_ns/chatter'],
+            fixture_actions=[get_talker_node_action()],
+            expected_output=itertools.cycle([
+                re.compile(r'data: Hello...'),
+                '---'
+            ]),
+            output_filter=basic_output_filter(
+                filtered_rmw_implementation=rmw_implementation
+            ),
+            self_terminates=False,
+            exit_codes=[2],
+            timeout=5.0,
+        ),
+        CLITestConfiguration(
+            command='topic',
+            arguments=['pub', '/my_ns/chatter', 'std_msgs/msg/String', '{data: test}'],
```

Comment:
Should the test also subscribe to the topic and check that the message is received?
I think that checking output is not enough in this case

Reply:
That's what we do with the listener and we then check it's output. Yes, we're trusting the listener in the process...